README.md

```diff
@@ -41,6 +41,74 @@ stack haddock
 
 ```
 
+## Quick-Start Example - File Uploader
+
+### FileUploader.hs
+``` haskell
+#!/usr/bin/env stack
+-- stack --resolver lts-6.27 runghc --package minio-hs --package optparse-applicative --package filepath
+
+{-# Language OverloadedStrings #-}
+import           Network.Minio
+
+import qualified Control.Monad.Catch as MC
+import           Control.Monad.IO.Class (liftIO)
+import qualified Data.Conduit.Combinators as CC
+import qualified Data.Text as T
+import           Options.Applicative
+import           Prelude
+import           System.FilePath.Posix
+
+-- | The following example uses minio's play server at
+-- https://play.minio.io:9000.  The endpoint and associated
+-- credentials are provided via the libary constant,
+--
+-- > minioPlayCI :: ConnectInfo
+--
+
+-- optparse-applicative package based command-line parsing.
+fileNameOpts :: Parser FilePath
+fileNameOpts = strOption
+               (long "filename"
+                <> metavar "FILENAME"
+                <> help "Name of file to upload to AWS S3 or a Minio server")
+
+cmdParser = info
+            (helper <*> fileNameOpts)
+            (fullDesc
+             <> progDesc "fileUploader"
+             <> header
+             "fileUploader - a simple file-uploader program using minio-hs")
+
+
+main :: IO ()
+main = do
+  let bucket = "my-bucket"
+
+  -- Parse command line argument, namely --filename.
+  filepath <- execParser cmdParser
+  let object = T.pack $ takeBaseName filepath
+
+  res <- runResourceT $ runMinio minioPlayCI $ do
+    -- Make a bucket; catch bucket already exists exception if thrown.
+    (MC.catch :: Minio () -> (MError -> Minio ()) -> Minio ())
+      (makeBucket bucket Nothing)
+      (\_ -> liftIO $ print "Bucket already exists, proceeding with upload file.")
```

Comment:
The exception handler doesn't specifically catch `BucketExists` exception alone. The library needs to provide a mechanism to identify specific S3 errors. One approach is to provide combinators for all the common S3 [error codes](http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList).
E.g,
```haskell
-- rethrows exception if error code doesn't match "BucketAlreadyExists".
errBucketAlreadyExists :: MError -> Bool
```